next/src/field/type.ts

```diff
@@ -1,47 +1,164 @@
- */
-export interface Field {
-  name: string
-  label?: string
-  description?: string
-  fields?: Field[]
-  // @deprecated in favor of inputType,
+export type FieldType = 'text' | 'number' | 'select' | 'file' | 'radio' | 'group-array' | 'email' | 'date' | 'checkbox' | 'fieldset' | 'money' | 'country' | 'textarea'
+
+interface BaseField {
   type: FieldType
-  inputType: FieldType
+  name: string
+  label: string
   required: boolean
+  inputType: FieldType
   jsonType: JsfSchemaType
+  errorMessage: Record<string, string>
+  schema: any
   isVisible: boolean
-  accept?: string
-  errorMessage?: Record<string, string>
-  computedAttributes?: Record<string, unknown>
+}
+
+export interface FieldOption {
+  label: string
+  value: string
+  description?: string
+}
+
+export interface FieldSelect extends BaseField {
+  type: 'select'
+  options: FieldOption[]
+}
+
+export interface FieldTextarea extends BaseField {
+  type: 'textarea'
+  maxLength?: number
+  minLength?: number
+}
+
+export interface FieldDate extends BaseField {
+  type: 'date'
+  format: string
   minDate?: string
   maxDate?: string
   maxLength?: number
-  maxFileSize?: number
-  format?: string
-  anyOf?: unknown[]
-  options?: unknown[]
-  const?: unknown
-  checkboxValue?: unknown
+}
 
-  // Allow additional properties from x-jsf-presentation (e.g. meta from oneOf/anyOf)
-  [key: string]: unknown
+export interface FieldText extends BaseField {
+  type: 'text'
+  maxLength?: number
+  maskSecret?: number
+  pattern?: string
 }
 
-/**
- * Field option
- * @description
- * Represents a key/value pair that is used to populate the options for a field.
- * Will be created from the oneOf/anyOf elements in a schema.
- */
-export interface FieldOption {
+export interface FieldRadio extends BaseField {
+  type: 'radio'
+  options: FieldOption[]
+  direction?: 'row' | 'column'
+  const?: string
+}
+
+export interface FieldNumber extends BaseField {
+  type: 'number'
+  minimum?: number
+  maximum?: number
+}
+
+export interface FieldMoney extends BaseField {
+  type: 'money'
+  currency: string
+}
+
+export interface FieldCheckbox extends BaseField {
+  type: 'checkbox'
+  options?: FieldOption[]
+  multiple?: boolean
+  direction?: 'row' | 'column'
+  checkboxValue?: string | boolean
+  const?: string
+}
+
+export interface FieldEmail extends BaseField {
+  type: 'email'
+  maxLength?: number
+  format: 'email'
+}
+
+export interface FieldFile extends BaseField {
+  type: 'file'
+  accept: string
+  multiple?: boolean
+  fileDownload: string
+  fileName: string
+}
+export interface FieldFieldSet extends BaseField {
+  type: 'fieldset'
+  valueGroupingDisabled?: boolean
+  visualGroupingDisabled?: boolean
+  variant?: 'card' | 'focused' | 'default'
+  fields: Field[]
+}
+
+export interface GroupArrayField extends BaseField {
+  type: 'group-array'
+  name: string
   label: string
-  value: unknown
-  [key: string]: unknown
+  description: string
+  fields: () => Field[]
```

Comment:
bug: This is v0, in v1 is `fields: Field[]` (Once https://github.com/remoteoss/json-schema-form/pull/177 is merged)) :D